frontend/templates/product-list/MetaTags.tsx

```diff
@@ -36,16 +37,21 @@ export function MetaTags({ productList }: MetaTagsProps) {
       title += ` - Page ${page}`;
    }
    title += ' | iFixit';
-   const itemTypeHandle = itemType ? `/${encodeDeviceItemType(itemType)}` : '';
-   const canonicalUrl = `${appContext.ifixitOrigin}${
-      productList.path
-   }${itemTypeHandle}${page > 1 ? `?${PRODUCT_LIST_PAGE_PARAM}=${page}` : ''}`;
+   const itemTypeHandle = itemType
+      ? `/${encodeDeviceItemTypeSpaces(itemType)}`
+      : '';
+   const deviceAndItemTypePath = encodeDevicePath(
+      `${productList.path}${itemTypeHandle}`
```

Comment:
I think this is a pretty odd bit of code and seeing it would make me question a bunch of surrounding code. If we know we have pieces of a url that need to be url-encoded, we should encode them *before* building a url. i.e.

`path = encodeUriComponent(piece1) + "/" + encodeUriComponent(piece2);`

Reply:
I agree. We already had encodeDeviceItemType and encodeDeviceTitle, so why make a new function that splits them apart and then encodes them?

Reply:
the `encodeDeviceItemType` and `encodeDeviceTitle` were not actually url encoding they were simply replacing ' ' with '_'. And theyre being used all over the place to basically just do that, and thats used in things like `const noIndexExemptions` which i wasn't sure if we wanted to change its readability to making all the keys url encoded, especially when it seems unnecessary. Which is why theres a commit to change the names so they are more accurate to what theyre doing.

As for why i join them and then encode, its because the path contains the entire base route so like `/Parts/MacBook_Air_13"` would be route. So we have to split it anyway, because encoding that will ruin the `/`s and break the route. Which is why originally i was using `encodeURI()`. And if we have to split them anyway and do the same exact thing to both the `Path` and `Item Type` I don't see a disadvantage to condensing that to one call.

Reply:
I think if there are places we're using encodeDevice* in a way that doesn't work with urlencoding, we should take another look and maybe remove that. I'm not sure why we would want to encode it partway. For example, in the noIndexExemptions, we should store the decoded values so we don't have to encode before looking it up. Then we can only encode immediately before creating a url, and we can encode the device and item type separately.